### PR TITLE
Support Apple M1 / darwin arm64 (closes #38)

### LINKED ISSUE #38: Support Apple M1 / darwin arm64
Feature request:

Support Apple M1 by additionally compiling and providing release binaries for darwin arm64

### PR DESCRIPTION
**Pull request by TiganeteaRobert**
Added GoReleaser action to `deploy.yml` in order to compile all binary types automatically.

### CODE REVIEW

**Comment by jbeemster** [comment]
@TiganeteaRobert you might be better of with https://github.com/goreleaser/goreleaser-action - this will compile every possible binary type in a single command.  You can see here how it works:

Action: https://github.com/snowplow-devops/terraform-provider-snowplow/blob/master/.github/workflows/cd.yml#L42-L51
Result: https://github.com/snowplow-devops/terraform-provider-snowplow/releases/tag/v0.6.0

Assuming you have a high enough Go Version in your build this will then produce every asset you could possibly need and might future proof things a touch!

**Comment by TiganeteaRobert** [comment]
@jbeemster I added GoReleaser and tested it, seems to work fine. Thanks a lot, it's a much cleaner solution!

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -34,27 +34,18 @@ jobs:
...
-    - name: Upload release binaries
-      uses: alexellis/upload-assets@0.2.3
+        version: latest
+        args: release --rm-dist
+        workdir: .
       env:
-        GITHUB_TOKEN: ${{ github.token }}
-      with:
-        asset_paths: '["./build/compiled/*"]'
+        GPG_FINGERPRINT: ${{ steps.import_gpg.outputs.fingerprint }}
```
Comment (jbeemster) [inline .github/workflows/deploy.yml:44]: Should this still be here?

Comment (TiganeteaRobert) [reply .github/workflows/deploy.yml:47]: Nope, nice catch! Removed.

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -33,28 +33,21 @@ jobs:
         key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
         restore-keys: |
           ${{ runner.os }}-go-
-
+
```
Comment (jbeemster) [inline .github/workflows/deploy.yml:36]: nit: random addition of whitespace?

File: Makefile [hunk 3]
```diff
@@ -16,13 +16,15 @@ output_dir   = $(build_dir)/output
...
 windows_out_dir = $(output_dir)/windows
 
-bin_name    = snowplow-tracking-cli
-bin_linux   = $(linux_out_dir)/$(bin_name)
-bin_darwin  = $(darwin_out_dir)/$(bin_name)
-bin_windows = $(windows_out_dir)/$(bin_name)
+bin_name       = snowplow-tracking-cli
+bin_linux      = $(linux_out_dir)/$(bin_name)
+bin_darwin     = $(darwin_out_dir)/$(bin_name)
+bin_arm 	   = $(arm_out_dir)/$(bin_name)
```
Comment (jbeemster) [inline Makefile:26]: nit: indents are off

**Review by jbeemster — COMMENTED** [review]
LGTM!  Would be worth tagging an rc to check the build process does work and the binaries produced by goreleaser function as expected.

**Comment by TiganeteaRobert** [comment]
@jbeemster already did, made a tag and release then deleted them.

File: Makefile (hunk 3 above)
Comment (TiganeteaRobert) [reply Makefile]: Weird, caused by tabs it seems. Fixed!

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (TiganeteaRobert) [reply .github/workflows/deploy.yml]: Removed.

**Comment by jbeemster** [comment]
Seems good to me @TiganeteaRobert - just need to rebase and cleanup the commit log and all the rest of the house-keeping now!

**Pull request merged**